tests/post_training/test_quantize_conformance.py

```diff
@@ -1,100 +1,181 @@
+import copy
+import logging
 import os
 import re
-import logging
+from abc import ABCMeta, abstractmethod
 from pathlib import Path
-import copy
+from typing import Callable
+
 import numpy as np
+import onnx
+import openvino.runtime as ov
 import pytest
-
 import torch
-import nncf
-import onnx
-from torchvision import datasets
-from torchvision import transforms
-from torchvision.transforms import InterpolationMode
-
-import timm
-from tqdm import tqdm
+from model_scope import get_validation_scope
 from sklearn.metrics import accuracy_score
+from utils import (create_timm_model, export_to_ir, export_to_onnx,
+                   get_model_transform, get_torch_dataloader, run_benchmark)
 
-import openvino.runtime as ov
+import nncf
+from nncf.common.utils.logger import logger as nncf_logger
 
-from tests.shared.command import Command
-from model_scope import get_validation_scope
+nncf_logger.setLevel(logging.ERROR)
 
 
 NOT_AVAILABLE_MESSAGE = "N/A"
 DEFAULT_VAL_THREADS = 4
+ILSVRC_VALIDATION_SUBSET_SIZE = 5000
 
 
-def create_timm_model(name):
-    model = timm.create_model(
-        name, num_classes=1000, in_chans=3, pretrained=True, checkpoint_path=""
-    )
-    return model
-
-
-def get_model_transform(model):
-    config = model.default_cfg
-    normalize = transforms.Normalize(mean=config["mean"], std=config["std"])
-    input_size = config["input_size"]
-    resize_size = tuple(int(x / config["crop_pct"]) for x in input_size[-2:])
-
-    RESIZE_MODE_MAP = {
-        "bilinear": InterpolationMode.BILINEAR,
-        "bicubic": InterpolationMode.BICUBIC,
-        "nearest": InterpolationMode.NEAREST,
-    }
-
-    transform = transforms.Compose(
-        [
-            transforms.Resize(
-                resize_size, interpolation=RESIZE_MODE_MAP[config["interpolation"]]
-            ),
-            transforms.CenterCrop(input_size[-2:]),
-            transforms.ToTensor(),
-            normalize,
-        ]
-    )
+class QuantizerBase(metaclass=ABCMeta):
+    """Base class for an executor"""
 
-    return transform
+    def __init__(self, name):
+        super().__init__()
+        self.name = name
 
+    @abstractmethod
+    def quantize(
```

Comment:
It would be nice to have a little fix for such code roughnesses if that's possible of course.